ha/setup/cluster_validator/cluster_status.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+Module to provide standalone interface to cluster status.
+"""
+
+import re
+from typing import Callable, List, NamedTuple, Union, Optional
+from xml.etree import ElementTree
+from xml.etree.ElementTree import Element
+from ha.core.error import SetupError, HA_CLUSTER_CONFIG_ERROR
+from ha.execute import SimpleCommand
+
+
+class ValidationStatusError(SetupError):
+    """Cluster status format is invalid."""
+    def __init__(self, desc=None):
+        _desc = "Error when parsing cluster status" if desc is None else desc
+        _message_id = HA_CLUSTER_CONFIG_ERROR
+        _rc = 1
+        super().__init__(rc=_rc, desc=_desc, message_id=_message_id)
+
+
+ClusterSummary = NamedTuple("ClusterSummary", [
+    ("maintenance_mode", bool),
+    ("quorum", bool),
+    ("stonith_enabled", bool),
+    ("num_nodes", int),
+    ("num_resources_configured", int),
+    ("num_resources_disabled", int)
+])
+
+ClusterNode = NamedTuple("ClusterNode", [
+    ("name", str),
+    ("online", bool),
+    ("standby", bool),
+    ("maintenance", bool),
+    ("unclean", bool),
+    ("resources_running", int),
+])
+
+ClusterResource = NamedTuple("ClusterResource", [
+    ("name", str),
+    ("clone", bool),
+    ("resource_agent", str),
+    ("role", str),  # Enum may be needed here
+    ("enabled", bool),
+    ("failed", bool),
+    ("managed", bool),
+    ("failure_ignored", bool),
+    ("location", List[str]),
```

Comment:
Looks like there is no real need to have locations as a list. This is obsolete and not required anymore.
(Note: this issue can be addressed in separate PR)